src/codegen/mod.rs

```diff
@@ -3560,6 +3591,29 @@ mod utils {
         result.extend(old_items.into_iter());
     }
 
+    pub fn prepend_block_header(
+        ctx: &BindgenContext,
+        result: &mut Vec<quote::Tokens>,
+    ) {
+        let use_block = if ctx.options().block_extern_crate {
+            quote! {
+                #![allow(improper_ctypes)]
```

Comment:
Please don't generate this attribute. There's no guarantee that bindings will be in a module on their own. We don't do it anywhere else.

Reply:
Perhaps, but that won't get executed unless you enable C++ namespaces. I'd leave the responsibility of ignoring the warning to the bindgen consumer.

Reply:
ok